### PR TITLE
Add drop_packets/vrf/hash/sub_port/gcu/span data plane test to onboarding PR test

### PR DESCRIPTION
**Pull request by xwjiang-ms**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
Elastictest performs well in distribute running PR test in multiple KVMs, which support us to add more test scripts to PR checker.
But some traffic test using ptfadapter can't be tested on KVM platform, we need to skip traffic test if needed
#### How did you do it?
Add drop_packets/vrf/hash/sub_port/gcu/span data plane test to onboarding PR test and skip traffic test

### CODE REVIEW

File: tests/common/plugins/conditional_mark/tests_mark_conditions.yaml [hunk 1]
```diff
@@ -721,21 +721,27 @@ hash/test_generic_hash.py::test_ecmp_and_lag_hash:
 
 hash/test_generic_hash.py::test_lag_member_flap:
   skip:
-    reason: 'On Mellanox SPC1 platforms, due to HW limitation, it would not support CRC_CCITT algorithm'
+    reason: 'On Mellanox SPC1 platforms, due to HW limitation, it would not support CRC_CCITT algorithm, for other platforms, skipping until the issue is addressed'
```
Comment (opcoder0) [inline tests/common/plugins/conditional_mark/tests_mark_conditions.yaml:724]: Can you please clarify the reason a bit more. I understand it is skipped on Mellanox SPC1. The new comment says it is skipped on other platforms while the skip here is only related to `vs`.

File: tests/hash/test_generic_hash.py [hunk 2]
```diff
@@ -357,21 +373,23 @@ def test_nexthop_flap(duthost, tbinfo, ptfhost, fine_params, mg_facts, restore_i
...
-            "generic_hash_test.GenericHashTest",
-            platform_dir="ptftests",
-            params=ptf_params,
-            log_file=PTF_LOG_PATH,
-            qlen=PTF_QLEN,
-            socket_recv_size=16384,
-            is_python3=True
-        )
+
+    if not skip_traffic_test:
```
Comment (opcoder0) [inline tests/hash/test_generic_hash.py:377]: Can we invert the condition and return like in other tests ? Like -

```
if skip_traffic_test is True:
        logger.info("Skipping traffic test")
        return
```

File: tests/vrf/vrf_neigh.j2 [hunk 3]
```diff
@@ -7,9 +7,9 @@
 {% endif -%}
 {%- endmacro %}
 
-{% for intf, ip_facts in intf_ips.iteritems() -%}
+{% for intf, ip_facts in intf_ips.items() -%}
```
Comment (opcoder0) [inline tests/vrf/vrf_neigh.j2:10]: Can you please confirm if these changes related to skipping traffic test ? I am not sure if the `iteritems` is not valid anymore ?

**Review by opcoder0 — COMMENTED** [review]
I see the changes are similar across files. Would it be possible to split this PR into multiple PRs because it touches too many files? I am not sure if every change uses `do_test`.

**Comment by xwjiang-ms** [comment]
> I see the changes are similar across files. Would it be possible to split this PR into multiple PRs because it touches too many files? I am not sure if every change uses `do_test`.

Yes agreed, I will create separate PRs

File: tests/vrf/vrf_neigh.j2 (hunk 3 above)
Comment (opcoder0) [reply tests/vrf/vrf_neigh.j2:10]: On a different PR I had a comment about keeping these changes compatible for backporting. So, can this instead be modified to support both Python 2 and 3 using `six`?

**Pull request closed without merging**